### PR TITLE
refactor workflow validator

### PR DESCRIPTION
**Pull request by honnix**
# Hey, I just made a Pull Request!

## Description

Split `WorkflowValidator` into basic and extra part.

Basic can be used by both client and server while extra can only be used by server.

## Motivation and Context

Action based on `null` check is hard to follow and is error prone.

## Checklist for PR author(s)

- [x] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java [hunk 1]
```diff
@@ -20,168 +20,24 @@
...
-        e.add("secret " + secret.name() + " is not whitelisted");
-      }
-    });
-
-    return e;
-  }
-
-  private <T extends Comparable<T>> void lowerLimit(List<String> errors, T value, T limit, String message) {
-    limit(errors,value.compareTo(limit) < 0, value, limit, message);
+  default <T extends Comparable<T>> void lowerLimit(List<String> errors, T value, T limit, String message) {
```
Comment (narape) [inline styx-common/src/main/java/com/spotify/styx/util/WorkflowValidator.java]: I see this methods as auxiliary methods and not part of the interface. So it would be better to have it as statics methods instead of default methods

File: styx-service-common/src/test/java/com/spotify/styx/util/ExtraWorkflowValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import java.util.List;
+import java.util.Set;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ExtraWorkflowValidatorTest {
```
Comment (narape) [inline styx-service-common/src/test/java/com/spotify/styx/util/ExtraWorkflowValidatorTest.java]: Add a test for making sure that the we are propagating a call to the delegate. For example that the validate method was called and that we propagate any error returned by the delegate

**Comment by honnix** [comment]
@narape PTAL again. I have rebased from master.
